src/config_format/flb_cf_yaml.c

```diff
@@ -880,417 +1111,405 @@ static int consume_event(struct flb_cf *cf, struct local_ctx *ctx,
+            }
             break;
         case YAML_SEQUENCE_END_EVENT:
-            if (s->section == SECTION_CUSTOM) {
-                /* 'customs' is a top level. So we get back to 'section'. */
-                s->state = STATE_SECTION;
-            }
-            else {
-                s->state = STATE_PIPELINE;
+            state = state_pop(ctx);
+            if (state == NULL) {
+                flb_error("no state left");
+                return YAML_FAILURE;
             }
             break;
         default:
-            yaml_error_event(ctx, s, event);
+            yaml_error_event(ctx, state, event);
             return YAML_FAILURE;
         }
         break;
 
-    case STATE_PLUGIN_KEY:
+    case STATE_PLUGIN_VAL:
         switch(event->type) {
         case YAML_SCALAR_EVENT:
-            s->state = STATE_PLUGIN_VAL;
-            value = (char *) event->data.scalar.value;
-            s->key = flb_sds_create(value);
+            /* register key/value pair as a property */
+            if (cfl_kvlist_insert_string(state->keyvals, state->key, (char *)event->data.scalar.value) < 0) {
+                flb_error("unable to insert string");
+                return YAML_FAILURE;
+            }
+            state = state_pop(ctx);
+            if (state == NULL) {
+                flb_error("no state left");
+                return YAML_FAILURE;
+            }
+            break;
+        case YAML_SEQUENCE_START_EVENT: /* start a new group */
+            state = state_push_witharr(ctx, state, STATE_PLUGIN_VAL_LIST);
+            if (state == NULL) {
+                flb_error("unable to allocate state");
+                return YAML_FAILURE;
+            }
             break;
         case YAML_MAPPING_START_EVENT:
-            s->state = STATE_PLUGIN_TYPE;
+            /* Special handling for input processor */
+            if (strcmp(state->key, "processors") == 0) {
+                state = state_push(ctx, STATE_INPUT_PROCESSORS);
+                if (state == NULL) {
+                    flb_error("unable to allocate state");
+                    return YAML_FAILURE;
+                }
+                state_create_group(cf, state, "processors");
+                break;
+            }
+
+            state = state_push(ctx, STATE_GROUP_KEY);
+            if (state == NULL) {
+                flb_error("unable to allocate state");
+                return YAML_FAILURE;
+            }
+            /* create group */
+            state->values = flb_cf_section_property_add_list(cf,
+                                                             state->cf_section->properties,
+                                                             state->key, flb_sds_len(state->key));
+            if (state->values == NULL) {
+                flb_error("no values");
+                return YAML_FAILURE;
+            }
+            state->cf_group = flb_cf_group_create(cf, state->cf_section, state->key, strlen(state->key));
```

Comment:
This result is not checked and inside of `flb_cf_group_create` in line there's a `cfl_kvlist_create` call that's not checked either so please fix it and add the appropriate disposal of `g->properties` in line 461 to fix the leak that happens when line 459 fails.